### PR TITLE
Add assert_precondition to the lint

### PR DESCRIPTION
**Pull request by stephenmcgruer**
This is deprecated in favour of assert_implements and
assert_implements_optional. There are a few files left using
assert_precondition, added to the lint ignorelist.

### CODE REVIEW

File: lint.whitelist [hunk 1]
```diff
@@ -756,3 +756,12 @@ TESTHARNESS-IN-OTHER-TYPE: svg/extensibility/foreignObject/foreign-object-circul
 TESTHARNESS-IN-OTHER-TYPE: svg/extensibility/foreignObject/foreign-object-under-clip-path-crash.html
 TESTHARNESS-IN-OTHER-TYPE: svg/extensibility/foreignObject/foreign-object-under-defs-crash.html
 TESTHARNESS-IN-OTHER-TYPE: svg/svg-in-svg/svg-in-svg-circular-filter-reference-crash.html
+
+# Temporary inclusions while we deprecate assert_precondition
```
Comment (Hexcles) [inline lint.whitelist:760]: Nit: maybe add TODO/FIXME for easier grep.

File: tools/lint/rules.py [hunk 2]
```diff
@@ -478,3 +478,12 @@ class PromiseRejectsRegexp(Regexp):
     file_extensions = [".html", ".htm", ".js", ".xht", ".xhtml", ".svg"]
     description = "Test-file line has a `promise_rejects(...)` call"
     to_fix = """Replace with promise_rejects_dom or promise_rejects_js or `promise_rejects_exactly`"""
+
+
+class AssertPreconditionRegexp(Regexp):
+    pattern = br"[^.]assert_precondition\("
+    name = "ASSERT-PRECONDITION"
```
Comment (Hexcles) [inline tools/lint/rules.py:485]: Use underscores for consistency.

```suggestion
    name = "ASSERT_PRECONDITION"
```

**Review by Hexcles — APPROVED** [review]
LGTM % nits

File: tools/lint/rules.py (hunk 2 above)
Comment (stephenmcgruer) [reply tools/lint/rules.py:485]: I don't care so am happy to change, but fyi rules.py uses hyphens, underscores, and even just spaces for different name entries. There's no consistent separator.

Comment (Hexcles) [reply tools/lint/rules.py:485]: Grrh... Never mind then. I only expanded a few lines above and saw `ASSERT_THROWS`.

**Review by jgraham — APPROVED** [review]

**Comment by Hexcles** [comment]
> ERROR:lint:workers/modules/dedicated-worker-parse-error-failure.html:27: Test-file line has an `assert_precondition(...)` call (ASSERT-PRECONDITION)

**Comment by stephenmcgruer** [comment]
Azure Pipelines failure seems unrelated, merging.

**Pull request merged**